lib/src/main/java/com/diffplug/spotless/FormatterProperties.java

```diff
@@ -0,0 +1,303 @@
+ */
+package com.diffplug.spotless;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/** Utility manages settings of formatter configured by properties. */
+public class FormatterProperties {
+
+	private static final Logger LOG = Logger.getLogger(FormatterProperties.class.getName());
+
+	private final Properties properties;
+
+	/**
+	 * Per default the property list is empty and the formatters default
+	 * settings are used.
+	 */
+	public FormatterProperties() {
+		properties = new Properties();
+	}
+
+	/**
+	 * Import settings from a sequence of files (file import is the given order)
+	 *
+	 * @param settingsFiles
+	 *            Sequence of files
+	 * @throws IllegalArgumentException
+	 *            In case the import of a file fails
+	 */
+	public void add(final Iterable<File> settingsFiles) throws IllegalArgumentException {
+		settingsFiles.forEach(file -> add(file));
+	}
+
+	/**
+	 * Import settings from given file. New settings (with the same ID/key)
+	 * override existing once.
+	 *
+	 * @param settingsFile
+	 *            File
+	 * @throws IllegalArgumentException
+	 *            In case the import of the file fails
+	 */
+	public void add(final File settingsFile) throws IllegalArgumentException {
+		if (!(settingsFile.isFile() || settingsFile.canRead())) {
+			String msg = String.format("Settings file '%s' does not exist or can not be read.", settingsFile);
+			throw new IllegalArgumentException(msg);
+		}
+		try {
+			Properties newSettings = FileParser.parse(settingsFile);
+			properties.putAll(newSettings);
+		} catch (IOException | IllegalArgumentException | NullPointerException exception) {
+			StringBuffer message = new StringBuffer(
+					String.format("Failed to add properties from '%s' to formatter settings.", settingsFile));
+			String detailedMessage = exception.getMessage();
+			if (null != detailedMessage) {
+				message.append(String.format(" %s", detailedMessage));
+			}
+			throw new IllegalArgumentException(message.toString(), exception);
+		}
+	}
+
+	/**
+	 * Get accumulated {@link java.util.Properties Properties}
+	 * @return Properties
+	 */
+	public Properties getProperties() {
+		return properties;
+	}
+
+	/**
+	 * Returns a {@link java.util.Set Set} view for the underlying property map.
+	 *
+	 * @return Property set
+	 */
+	public Set<Map.Entry<Object, Object>> entrySet() {
+		return properties.entrySet();
+	}
```

Comment:
Because if you _are_ after a solution that produces something immutable and more type-safe, then `com.diffplug.common.collect.Maps.fromProperties` would do the trick I think (assuming Diffplug's fork of Guava even has that class & method).